syntax = "proto3";
package pb;


message IndexBlock {
    repeated BlockOffset offsets = 1;
    bytes bloom_filter = 2;
    uint32 max_version = 3;
    uint32 key_count = 4;
}

message BlockOffset {
    string base_key = 1;
    uint64 offset = 2;
    uint32 len = 3; 
} 

message ManifestChangeSet {
    repeated ManifestChange changes = 1;
}

message ManifestChange {
    uint32 id = 1;
    enum Operation {
        CREATE = 0;
        DELETE = 1;
    }
    Operation op = 2;
    uint32 level = 3;
    uint32 check_sum = 4;
} 